library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/TextWireframeMapper.kt

```diff
@@ -9,28 +9,29 @@ package com.datadog.android.sessionreplay.internal.recorder.mapper
 import android.graphics.Typeface
 import android.view.Gravity
 import android.widget.TextView
+import com.datadog.android.sessionreplay.internal.recorder.SystemInformation
 import com.datadog.android.sessionreplay.internal.recorder.densityNormalized
 import com.datadog.android.sessionreplay.model.MobileSegment
 
 internal open class TextWireframeMapper :
     BaseWireframeMapper<TextView, MobileSegment.Wireframe.TextWireframe>() {
 
-    override fun map(view: TextView, pixelsDensity: Float):
+    override fun map(view: TextView, systemInformation: SystemInformation):
         List<MobileSegment.Wireframe.TextWireframe> {
-        val viewGlobalBounds = resolveViewGlobalBounds(view, pixelsDensity)
-        val styleBorderPair = view.background?.resolveShapeStyleAndBorder(view.alpha)
+        val viewGlobalBounds = resolveViewGlobalBounds(view, systemInformation.screenDensity)
+        val resolveShapeStyleAndBorder = view.background?.resolveShapeStyleAndBorder(view.alpha)
```

Comment:
probably can do simpler here. not sure if it works with nullable type for destructuring, otherwise can use shorter name `shapeStyleAndBorder`

```suggestion
        val (shapeStyle, border) = view.background?.resolveShapeStyleAndBorder(view.alpha)
```